### PR TITLE
Refactor hook naming convention and its usage

### PR DESCRIPTION
**Pull request by akanex115**
# Background
https://github.com/ajktown/wordnote/issues/44

## TODOs
- [x] Wait until https://github.com/ajktown/wordnote/pull/101 is merged
- [x] Wait until https://github.com/ajktown/wordnote/pull/159 is merged
- [x] Wait until https://github.com/ajktown/wordnote/pull/136 is merged
- [x] Use https://github.com/ajktown/wordnote/pull/159 as the standard
  - [x] Refactor every hook to have the prefix `on` for functions they return
  - [x] Refactor every component that takes the function from hook to keep the same name
  - [x] Double check if function with prefix `handle` is incorrectly used. 

## Checklist (Developers)
- [x] Make this PR as a draft first
- [x] Title is checked
- [x] Background & TODOs are filled
- [x] Assignee is set
- [x] Labels are set
- [x] Project is created & linked, if multiple PRs are associated to this PR
- [x] Appropriate issue(s) is(are) linked to this PR under `development`
- [x] `yarn inspect` is run
- [x] `TODOs` are handled and checked
- [x] Final Operation Check is done
- [x] Make this PR as an open PR

## Checklist (Reviewers)
- [x] Check if there are any other missing TODOs that are not yet listed
- [x] Review Code
- [x] Every item on the checklist has been addressed accordingly
- [x] If `development` is associated to this PR, you must check if every TODOs are handled

### CODE REVIEW

**Comment by mlajkim** [comment]
Seems okay to me? 
Can you first merge https://github.com/ajktown/wordnote/pull/159.

**Order of merge**
1.  https://github.com/ajktown/wordnote/pull/159
2. other refactor related wordnote PRs
3. this PR

File: src/hooks/preference/use-preference.hook.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { getPreferenceApi } from '@/api/preferences/get-preferences.api'
  * from the server and updates the preference state.
  */
 export const usePreference = () => {
-  const onClick = useRecoilCallback(
+  const onPreference = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/preference/use-preference.hook.ts:9]: @akanex115 ちょっと紛らわしいけど、
usePreference => GET => onGetPreference
usePutPreference => PUT => onPutPreference

File: src/hooks/semesters/use-apply-semester-details.ts [hunk 2]
```diff
@@ -16,5 +16,5 @@ export const useApplySemesterDetails = () => {
     [],
   )
 
-  return handleApplySemester
+  return onApplySemesterDetails
```
Comment (mlajkim) [inline src/hooks/semesters/use-apply-semester-details.ts:19]: @akanex115 
`onApplySemesterDetails`を返すから、これを使っている人も `onApplySemesterDetails`にしてほしい

src/hooks/words/use-word-ids.hook.tsの20行
src/hooks/words/use-words.hook.ts の21行

File: src/hooks/semesters/use-semester-by-code.hook.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { getSemesterByCodeApi as getSemesterByCodeApi } from '@/api/semesters/ge
 import { semesterDetailsFamily } from '@/recoil/words/semesters.state'
 
 export const useSemesterByCode = () => {
-  const handleRefresh = useRecoilCallback(
+  const onSemesterByCode = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/semesters/use-semester-by-code.hook.ts:6]: onGetSemesterByCode

File: src/hooks/semesters/use-semesters.hook.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { useRecoilCallback } from 'recoil'
 
 type HandleRefresh = () => Promise<GetSemestersResDTO>
 export const useSemesters = () => {
-  const handleRefresh: HandleRefresh = useRecoilCallback(
+  const onSemesters: HandleRefresh = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/semesters/use-semesters.hook.ts:10]: onGetSemsters

File: src/hooks/use-open-new-tab.ts [hunk 5]
```diff
@@ -2,10 +2,10 @@ import { useCallback } from 'react'
 
 /** Returns onClick that will open a new tab of your browser with the given link */
 export const useOpenNewTab = (link: string) => {
-  const onClick = useCallback(() => {
+  const onOpenNewTab = useCallback(() => {
```
Comment (mlajkim) [inline src/hooks/use-open-new-tab.ts:5]: これも名前はいいけど、使っているcomponentもonClickじゃなく、あわせよ

File: src/hooks/use-refresh.hook.ts [hunk 6]
```diff
@@ -9,9 +9,9 @@ type UseHookRefresh = [
 export const useHookRefresh = (): UseHookRefresh => {
   const [internalFlag, setInternalFlag] = useState(false)
 
-  const handleRefresh = useCallback(() => {
+  const onHookRefresh = useCallback(() => {
```
Comment (mlajkim) [inline src/hooks/use-refresh.hook.ts:12]: onRefresh?

File: src/hooks/words/use-word-by-id.hook.ts [hunk 7]
```diff
@@ -6,7 +6,7 @@ import { useApiErrorHook } from '../use-api-error.hook'
 export const useWordById = (id: string) => {
   const handleApiError = useApiErrorHook()
 
-  const handleRefresh = useRecoilCallback(
+  const onWordById = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/words/use-word-by-id.hook.ts:9]: onGetWordById

File: src/hooks/words/use-words-with-semesters.hook.ts [hunk 8]
```diff
@@ -15,7 +15,7 @@ export const useWordsWithSemesters = () => {
   const [, onGetWords] = useWords()
   const onGetSemesters = useSemesters()
 
-  const onGetWordsWithSemesters = useCallback(async () => {
+  const onWordWithSemesters = useCallback(async () => {
```
Comment (mlajkim) [inline src/hooks/words/use-words-with-semesters.hook.ts:18]: onGetWordWithSemesters

File: src/hooks/words/use-words.hook.ts [hunk 9]
```diff
@@ -20,7 +20,7 @@ export const useWords = (): UseWordIds => {
   const handleApiError = useApiErrorHook()
   const handleApplySemesterDetails = useApplySemesterDetails()
 
-  const handleRefresh: HandleRefresh = useRecoilCallback(
+  const onWords: HandleRefresh = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/words/use-words.hook.ts:23]: onGetWords

**Review by mlajkim — COMMENTED** [review]
Hookを使っているやつらも名前同一にしてくださいん

File: src/hooks/use-open-new-tab.ts (hunk 5 above)
Comment (akanex115) [reply src/hooks/use-open-new-tab.ts:5]: Handled in [36b7cf8](https://github.com/ajktown/wordnote/pull/158/commits/36b7cf8ac0f01d8984bc132fa24649e9d87ed42e)

File: src/hooks/words/use-words.hook.ts (hunk 9 above)
Comment (akanex115) [reply src/hooks/words/use-words.hook.ts]: Handled in [d98b17b](https://github.com/ajktown/wordnote/pull/158/commits/d98b17b1c2670af41b7b0116430834246b37fd00)

File: src/hooks/preference/use-preference.hook.ts (hunk 1 above)
Comment (akanex115) [reply src/hooks/preference/use-preference.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/semesters/use-semester-by-code.hook.ts (hunk 3 above)
Comment (akanex115) [reply src/hooks/semesters/use-semester-by-code.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/semesters/use-semesters.hook.ts (hunk 4 above)
Comment (akanex115) [reply src/hooks/semesters/use-semesters.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/use-refresh.hook.ts (hunk 6 above)
Comment (akanex115) [reply src/hooks/use-refresh.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/words/use-word-by-id.hook.ts (hunk 7 above)
Comment (akanex115) [reply src/hooks/words/use-word-by-id.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/words/use-words-with-semesters.hook.ts (hunk 8 above)
Comment (akanex115) [reply src/hooks/words/use-words-with-semesters.hook.ts]: Handled in [5ddf33e](https://github.com/ajktown/wordnote/pull/158/commits/5ddf33e05136c5e01ce02ca05840754462e6215c)

File: src/hooks/words/use-word-ids.hook.ts [hunk 10]
```diff
@@ -17,9 +17,9 @@ type UseWordIds = [boolean, HandleRefresh]
 export const useWordIds = (): UseWordIds => {
   const [loading, setLoading] = useState(false)
   const handleApiError = useApiErrorHook()
-  const handleApplySemesterDetails = useApplySemesterDetails()
+  const onApplySemesterDetails = useApplySemesterDetails()
 
-  const handleRefresh: HandleRefresh = useRecoilCallback(
+  const onWordIds: HandleRefresh = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/words/use-word-ids.hook.ts:22]: onWordIds => onGetWordIds
HandleRefresh => OnGetWordIds (最初の文字が大文字)

File: src/hooks/use-refresh.hook.ts [hunk 11]
```diff
@@ -9,9 +9,9 @@ type UseHookRefresh = [
 export const useHookRefresh = (): UseHookRefresh => {
   const [internalFlag, setInternalFlag] = useState(false)
```
Comment (mlajkim) [inline src/hooks/use-refresh.hook.ts:11]: Can you also write a TODO that 
this file is not used.

File: src/hooks/use-api-error.hook.ts [hunk 12]
```diff
@@ -6,20 +6,20 @@ import { useOnSignOutApp } from './app/use-on-sign-out-app.hook'
 type HandleApiError = (err: unknown) => any
 
 export const useApiErrorHook = (): HandleApiError => {
-  const handleSignOutApp = useOnSignOutApp()
+  const onAPiErrorHook = useOnSignOutApp()
 
-  const handleApiError: HandleApiError = useRecoilCallback(
+  const onApiErrorHook: HandleApiError = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/use-api-error.hook.ts:11]: Make sure that the type has the same name

HandleApiError => OnApiErrorHook

File: src/hooks/semesters/use-semesters.hook.ts [hunk 13]
```diff
@@ -7,7 +7,7 @@ import { useRecoilCallback } from 'recoil'
 
 type HandleRefresh = () => Promise<GetSemestersResDTO>
 export const useSemesters = () => {
-  const onSemesters: HandleRefresh = useRecoilCallback(
+  const onGetSemesters: HandleRefresh = useRecoilCallback(
```
Comment (mlajkim) [inline src/hooks/semesters/use-semesters.hook.ts:10]: HandleRefresh => OnGetSemesters

**Review by mlajkim — COMMENTED** [review]
impressive! second review done

File: src/hooks/words/use-word-ids.hook.ts (hunk 10 above)
Comment (akanex115) [reply src/hooks/words/use-word-ids.hook.ts]: Sorry for splitting it into two commits.
Handled in [refactor: onWordIds => onGetWordIds](https://github.com/ajktown/wordnote/pull/158/commits/6002906e7f370665866f91881e8713e76205ea59)
Handled in [refactor: HandleRefresh => OnGetWordIds](https://github.com/ajktown/wordnote/pull/158/commits/0122694a579fb4b7083dc6b0746c191d8431bcb8)

File: src/hooks/use-refresh.hook.ts (hunk 11 above)
Comment (akanex115) [reply src/hooks/use-refresh.hook.ts]: Handled in [718ff20](https://github.com/ajktown/wordnote/pull/158/commits/718ff206cc67382bbdc1ccb45fd89aff91953879)

File: src/hooks/use-api-error.hook.ts (hunk 12 above)
Comment (akanex115) [reply src/hooks/use-api-error.hook.ts]: Handled in [6015b91](https://github.com/ajktown/wordnote/pull/158/commits/6015b9172668f39d6487836b62c2d30bdb1bf421)

File: src/hooks/semesters/use-semesters.hook.ts (hunk 13 above)
Comment (akanex115) [reply src/hooks/semesters/use-semesters.hook.ts]: Handled in [6015b91](https://github.com/ajktown/wordnote/pull/158/commits/6015b9172668f39d6487836b62c2d30bdb1bf421)

**Review by mlajkim — APPROVED** [review]
OK

**Pull request merged**